c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandler.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    private C2OperationAck operationAck(C2Operation operation, C2OperationState state) {
+        C2OperationAck operationAck = new C2OperationAck();
+        operationAck.setOperationId(ofNullable(operation.getIdentifier()).orElse(EMPTY));
+        operationAck.setOperationState(state);
+        return operationAck;
+    }
+
+    private C2OperationState operationState(OperationState operationState, String details) {
+        C2OperationState state = new C2OperationState();
+        state.setState(operationState);
+        state.setDetails(details);
+        return state;
+    }
+
+    private List<Path> filterContent(List<Path> bundleFilePaths) {
+        List<Path> contentFilteredFilePaths = new ArrayList<>();
+        for (Path path : bundleFilePaths) {
+            String fileName = path.getFileName().toString();
+            try (Stream<String> fileStream = lines(path)) {
+                Path tempDirectory = createTempDirectory(null);
+                Path tempFile = Paths.get(tempDirectory.toAbsolutePath().toString(), fileName);
+                write(tempFile, (Iterable<String>) fileStream.filter(contentFilter)::iterator);
+                contentFilteredFilePaths.add(tempFile);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        return contentFilteredFilePaths;
+    }
+
+    private Optional<byte[]> createDebugBundle(List<Path> filePaths) {
+        ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+        try (GzipCompressorOutputStream gzipCompressorOutputStream = new GzipCompressorOutputStream(byteOutputStream);
+             TarArchiveOutputStream tarOutputStream = new TarArchiveOutputStream(gzipCompressorOutputStream)) {
+            for (Path filePath : filePaths) {
+                TarArchiveEntry tarArchiveEntry = new TarArchiveEntry(filePath.toFile(), filePath.getFileName().toString());
+                tarOutputStream.putArchiveEntry(tarArchiveEntry);
+                copy(filePath, tarOutputStream);
+                tarOutputStream.closeArchiveEntry();
+            }
+            tarOutputStream.finish();
+        } catch (Exception e) {
+            LOG.error("Error during create compressed bundle", e);
+            return empty();
+        } finally {
+            closeQuietly(byteOutputStream);
+        }
+        return Optional.of(byteOutputStream).map(ByteArrayOutputStream::toByteArray);
+    }
+
+    private void cleanup(List<Path> paths) {
+        Optional<Path> firstPath = paths.stream().findFirst();
+        if (firstPath.isPresent()) {
+            Path baseDirectory = firstPath.get().getParent();
+            paths.forEach(path -> deleteQuietly(path, "Unable to delete temporary file"));
+            deleteQuietly(baseDirectory, "Unable to delete temporary directory");
+        }
+    }
+
+    private static void deleteQuietly(Path baseDirectory, String errorMessage) {
```

Comment:
Is it intentional that this is static and the cleanup for example is not?

Reply:
Removed the static keyword (intellij generated it as static)